Tests/CommandsTests/TestCommandTests.swift

```diff
@@ -1172,4 +1172,263 @@ struct TestCommandTests {
+                    "got stdout: \(stdout), stderr: \(stderr)",
+                )
+            }
+        }
+
+        @Test(arguments: SupportedBuildSystemOnAllPlatforms)
+        func debuggerFlagWithXCTestSuite(buildSystem: BuildSystemProvider.Kind) async throws {
+            let configuration = BuildConfiguration.debug
+            try await fixture(name: "Miscellaneous/TestDebugging") { fixturePath in
+                let (stdout, stderr) = try await execute(
+                    ["--debugger", "--disable-swift-testing", "--verbose"],
+                    packagePath: fixturePath,
+                    configuration: configuration,
+                    buildSystem: buildSystem,
+                )
+
+                #expect(
+                    !stderr.contains("error: --debugger cannot be used with"),
+                    "got stdout: \(stdout), stderr: \(stderr)",
+                )
+
+                #expect(
+                    stdout.contains("target create") && stdout.contains("xctest"),
+                    "Expected LLDB to target xctest binary, got stdout: \(stdout), stderr: \(stderr)",
+                )
+
+                #expect(
+                    stdout.contains("failbreak breakpoint set -n \"_XCTFailureBreakpoint\""),
+                    "Expected XCTest failure breakpoint setup, got stdout: \(stdout), stderr: \(stderr)",
+                )
+            }
+        }
+
+        @Test(arguments: SupportedBuildSystemOnAllPlatforms)
+        func debuggerFlagWithSwiftTestingSuite(buildSystem: BuildSystemProvider.Kind) async throws {
+            let configuration = BuildConfiguration.debug
+            try await fixture(name: "Miscellaneous/TestDebugging") { fixturePath in
+                let (stdout, stderr) = try await execute(
+                    ["--debugger", "--disable-xctest", "--verbose"],
+                    packagePath: fixturePath,
+                    configuration: configuration,
+                    buildSystem: buildSystem,
+                )
+
+                #expect(
+                    !stderr.contains("error: --debugger cannot be used with"),
+                    "got stdout: \(stdout), stderr: \(stderr)",
+                )
+
+                #expect(
+                    stdout.contains("target create") && stdout.contains("swiftpm-testing-helper"),
+                    "Expected LLDB to target swiftpm-testing-helper binary, got stdout: \(stdout), stderr: \(stderr)",
+                )
+
+                #expect(
+                    stdout.contains("failbreak breakpoint set -s Testing -n \"failureBreakpoint()\""),
+                    "Expected Swift Testing failure breakpoint setup, got stdout: \(stdout), stderr: \(stderr)",
+                )
+            }
+        }
+
+        @Test(arguments: SupportedBuildSystemOnAllPlatforms)
+        func debuggerFlagWithBothTestingSuites(buildSystem: BuildSystemProvider.Kind) async throws {
```

Comment:
Would there be an easy way to actually interact with the LLDB during tests (eg,`stdin`)? For instance, send it some commands and check back the expected results? Maybe you did it already... and I missed :)

Reply:
One of the reasons this is still in draft is I want to address that. I think I'd have to add supplemental commands to the file that gets passed to LLDB that contains its setup commands. That would let us set specific breakpoints in tests, though I still haven't figured out how to inspect the output since we replace the running process with `execv`.